docs/users_guide/glasgow_exts.rst

```diff
@@ -4001,14 +4000,19 @@ For a full specification of the algorithms used in :ghc-flag:`-XDeriveFunctor`,
 :ghc-flag:`-XDeriveFoldable`, and :ghc-flag:`-XDeriveTraversable`, see
 :ghc-wiki:`this wiki page <Commentary/Compiler/DeriveFunctor>`.
 
-.. _deriving-typeable:
+.. _deriving-data:
 
-Deriving ``Typeable`` instances
+Deriving ``Data`` instances
```

Comment:
Here I'm adding a new section, "Deriving `Data` instances", for the sake of having a more appropriate place to mention the `DeriveDataTypeable` extension.